src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -524,12 +523,22 @@ def _init_monitored_items(self):
             "Valid_" + t for t in self.loss_logging_items_names + get_metrics_titles(self.valid_metrics)
         ]
 
+        # make sure the metric_to_watch is an exact match
+        metric_titles = self.loss_logging_items_names + get_metrics_titles(self.valid_metrics)
+        metric_to_watch_idx = fuzzy_idx_in_list(self.metric_to_watch, metric_titles)
+        metric_to_watch = metric_titles[metric_to_watch_idx]
+        if metric_to_watch != self.metric_to_watch:
+            logger.warning(
+                f"No exact match found for `metric_to_watch={self.metric_to_watch}`. It should be one of {metric_titles}. \n"
+                f"`metric_to_watch={metric_to_watch} will be used instead.`"
+            )
+            self.metric_to_watch = metric_to_watch
```

Comment:
Not mandatory, but I thought it is nice to let the user know that he has a typo, because we are not going to use the value of metric_to_watch  that he/she passed as an input and I don't think this should be silent.
But if you think it's too much (or that it would just be lost in the logs) we can take it out

Reply:
Sure, why not.